### PR TITLE
MDEV-14681 Bogus ER_UNSUPPORTED_EXTENSION

### PR DESCRIPTION
**Pull request by kevgs**

### CODE REVIEW

**Comment by midenok** [comment]
Reformat test case, make shorter.

**Comment by kevgs** [comment]
Updated.

File: sql/handler.cc [hunk 1]
```diff
@@ -3056,8 +3076,7 @@ int handler::update_auto_increment()
   enum enum_check_fields save_count_cuted_fields;
   DBUG_ENTER("handler::update_auto_increment");
 
-  // ALTER TABLE ... ADD COLUMN ... AUTO_INCREMENT
-  if (thd->lex->sql_command == SQLCOM_ALTER_TABLE)
+  if (is_alter_table_add_column_auto_increment(thd))
```
Comment (midenok) [inline sql/handler.cc]:
```c++
if (thd->lex->sql_command == SQLCOM_ALTER_TABLE && is_add_auto_increment(thd))
```

**Comment by kevgs** [comment]
Updated.

**Review by midenok — APPROVED** [review]

**Pull request merged**